gen/src/main/java/com/softlayer/api/gen/ClassWriter.java

```diff
@@ -377,8 +377,14 @@ public ClassWriter emitType() throws IOException {
         
         // Now the service
         if (!type.meta.noservice) {
-            if (type.meta.properties.containsKey("id")) {
-                if (type.meta.properties.containsKey("globalIdentifier")) {
+
+            Boolean containsId = type.meta.properties.containsKey("id")
+                    || type.baseMeta.properties.containsKey("id");
```

Comment:
Might as well make this recursive. Otherwise this only works one level deep. One easy way is to have `getProperty` on `TypeClass` that looks at parent if not local.